sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala

```diff
@@ -155,6 +155,16 @@ private[spark] object HiveUtils extends Logging {
     .booleanConf
     .createWithDefault(true)
 
+  val HIVE_TABLE_SCAN_MAX_PARALLELISM = buildConf("spark.sql.hive.tableScan.maxParallelism")
+    .doc("When reading Hive partitioned table, the default parallelism is the sum of Hive " +
+      "partition RDDs' parallelism. For Hive table of many partitions with many files, " +
+      "the parallelism could be very big and not good for Spark job scheduling. This optional " +
+      "config can set a maximum parallelism for reading Hive partitioned table. If the result " +
+      "RDD of reading such table is larger than this value, Spark will reduce the partition " +
+      "number by doing a coalesce on the RDD.")
+    .intConf
+    .createOptional
```

Comment:
Do we need a default value?

Reply:
The reason I leave it optional is to allow us keep current behavior.